docs/architecture/adt-014-proportional-slashing.md

```diff
@@ -0,0 +1,87 @@
+
+```
+slash_amount = (power_1 + power_2 + ... + power_n) // where power_i is the voting power of the ith validator faulting in the period
+```
+
+Now, if someone splits a validator of 10% into two validators of 5% each which both fault, then they both fault in the same window, they both will still get slashed at the sum 10% amount.
+
+However, an operator might still choose to split up their stake across multiple accounts with hopes that if any of them fault independently, they will not get slashed at the full amount.  In the case that the validators do fault together, they will get slashed the same amount as if they were one entity.  There is no con to splitting up.  However, if operators are going to split up their stake without actually decorrelating their setups, this also causes a negative externality to the network as it fills up validator slots that could have gone to others or increases the commit size.  In order to disincentivize this, we want it to be the case such that splitting up a validator into multiple validators and they fault together is punished more heavily that keeping it as a single validator that faults.
+
+We can achieve this by not only taking into account the sum of the percentages of the validators that faulted, but also the *number* of validators that faulted in the window.  One general form for an equation that fits this desired property looks like this:
+
+```
+slash_amount = (sqrt(power_1) + sqrt(power_2) + ... + sqrt(power_n))^2
+```
+
+So now, for example, if one validator of 10% faults, it gets a 10% slash, while if two validators of 5% each fault together, they both get a 20% slash ((sqrt(0.05)+sqrt(0.05))^2).
+
+One will note, that this model doesn't differentiate between multiple validators run by the same operators vs validators run by different operators.  This can be seen as an additional benefit in fact.  It incentivizes validators to differentiate their setups from other validators, to avoid having correlated faults with them or else they risk a higher slash.  So for example, operators should avoid using the same popular cloud hosting platforms or using the same Staking as a Service providers.  This will lead to a more resilient and decentralized network.
+
+
+We can allow some parameterization by multiplying the slash by an on-chain governable parameter.
+
+```
+slash_amount = k * (sqrt(power_1) + sqrt(power_2) + ... + sqrt(power_n))^2  // where k is an on-chain parameter for this specific slash type
+```
+
+This can be used to weight different types of slashes.  For example, we may want to punish liveness faults 10% as severely as double signs.  The k factor can also be something other than a constant, there is some research on using things like inverse gini coefficients to mitigate some griefing attacks, but this an area for future research.
```

Comment:
The exponent (`sqrt` / `^2`) should also be a parameter, and I am not sure if `2` is the optimal choice (why?)

Reply:
I don't really have a good justification for why sqrt and ^2 is correct from a theoretical side, it was just the first thing I tried.  But from an implementation side, I think it's the most efficient.  We have to find do this calculation in the state machine.  `sqrt` has a relatively fast algorithm to find it (Newton's method), I'm not sure if something as fast for a configurable root exists.

Reply:
Newton's method works for any function with a continuous derivative.

Are you thinking of the [Babylonian method](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Babylonian_method)?

Reply:
@cwgoes Do you have an idea of how to justify the exponent?

For parameterization, do you think this needs to be an on-chain governable parameter or just an in-code one?

Reply:
> @cwgoes Do you have an idea of how to justify the exponent? Not really sure, without any useful real data.

Yeah, I'm not sure either, but the value of the exponent affects the penalty for multiple correlated faults with power summing to some `x` vs a single fault for a validator with power `x`, maybe we should just mention that and mention that it could be changed.

> For parameterization, do you think this needs to be an on-chain governable parameter or just an in-code one?

For parameterisation, I think an on-chain governable parameter would be nice, since it's easy enough to change (no state should be dependent on it).

Reply:
Updated ADR to reflect that its an on-chain parameter.